backend/app/assets/stylesheets/spree/backend/shared/_tables.scss

```diff
@@ -4,27 +4,9 @@ table {
   border-spacing: 0;
 
   th, td {
-    border-right: 1px solid $color-border;
-    border-bottom: 1px solid $color-border;
-    vertical-align: middle;
+    vertical-align: top;
```

Comment:
I actually prefer the before to the after. As a general rule, I try to keep less space between the lines of the same label/list item/paragraph and more between it and whatever is around. It helps keep the item grouped with itself a little better.